src/Common/Auth/AuthUtils.php

```diff
@@ -967,4 +961,78 @@ private function clearFromMemory(&$password)
             $password = '';
         }
     }
+
+    /**
+     * Validates a google ID token and returns a user on success. If validation
+     * fails, return false.
+     *
+     * @param $token
+     * @return array|bool
+     */
+    static public function verifyGoogleSignIn($token)
+    {
+        $return = false;
+        $event = 'login';
+        $beginLog = 'Google Failure';
+        $ip = collectIpAddresses();
+        $client = new Google_Client([
+            'client_id' => $GLOBALS['google_signin_client_id']]);  // Specify the CLIENT_ID of the app that accesses the backend
+        $payload = $client->verifyIdToken($token);
```

Comment:
What is returned if the verifyIdToken call fails?
If it fails would error the log (note there will be no user or group; so would use `''` in the log entry for those) and return false. Then would do the following steps (realize different from what i stated above since the hook here is the google email and not the username as i mistakenly assumed above):
1. After the  getUserByGoogleSigninEmail (add LEFT JOIN to the query in that function: https://www.w3schools.com/sql/sql_join_left.asp and for the SELECT explicitly list the items you want including `U`.`username` as `user_username` and `US`.`username` as `usersecure_username`), then check the following sequentially. If fail, then log it and return false.
1a. If completely empty, then the user google email does not exist (log the email address in the comments and blank for the user and group)
1b. If not empty, but usersecure_username is not set, then the user login is not configured
1c. If not empty, but not active, then the user is not active
2. Check group, you do this already, but note this is not a acl group check (so change the logcomment)
3. Check to ensure user is in a ACL group: here is an example of how this is done in the user auth: https://github.com/openemr/openemr/blob/master/src/Common/Auth/AuthUtils.php#L302

Reply:
So I just replaced the join with a left join, then added the two select variables at the end of the select statement.  That should give us what we are looking for. I will post the code soon.

Reply:
hi @growlingflea ,
What does $payload get populated with if that auth fails (ie. if call to verifyIdToken() fails)?

Reply:
just answered my own question :)
https://github.com/googleapis/google-api-php-client/blob/2fb6e702aca5d68203fa737f89f6f774022494c6/src/Client.php#L773